src/vs/workbench/browser/parts/editor/textEditor.ts

```diff
@@ -156,12 +156,13 @@ export abstract class AbstractTextEditor<T extends IEditorViewState> extends Abs
 	}
 
 	protected getConfigurationOverrides(): ICodeEditorOptions {
+		const editorOptionsOverrideRaw = this.textResourceConfigurationService.getValue<ICodeEditorOptions>(this.getActiveResource(), 'editor');
 		return {
 			overviewRulerLanes: 3,
 			lineNumbersMinChars: 3,
 			fixedOverflowWidgets: true,
 			...this.getReadonlyConfiguration(this.input?.isReadonly()),
-			renderValidationDecorations: 'on' // render problems even in readonly editors (https://github.com/microsoft/vscode/issues/89057)
+			renderValidationDecorations: editorOptionsOverrideRaw.renderValidationDecorations // Originally defaulted 'on': render problems even in readonly editors (https://github.com/microsoft/vscode/issues/89057)
```

Comment:
I think this is a strange location to be using a configuration. The overrides are specifically to derive from configured settings. Should we remove this override then?

Reply:
Yeah, it seems like the override here is also specifically for `renderValidationDecorations` (since it was intended to just be `on` always.